packages/wrangler/src/deployment-bundle/create-worker-upload-form.ts

```diff
@@ -184,27 +184,6 @@ export function createWorkerUploadForm(worker: CfWorkerInit): FormData {
 		observability,
 	} = worker;
 
-	const assetConfig = {
-		html_handling: experimental_assets?.assetConfig?.html_handling,
-		not_found_handling: experimental_assets?.assetConfig?.not_found_handling,
-	};
-
-	// short circuit if static assets upload only
```

Comment:
I might be missing smth, so just for me to understand...why can't we keep this short-circuit and attach the `compatibility_date` to the metadata object? Aka

```
formData.set(
			"metadata",
			JSON.stringify({
                                 ...(compatibility_date && { compatibility_date }),
				assets: {
					jwt: experimental_assets.jwt,
					config: assetConfig,
				},
			})
		);
```

Reply:
It would work with the short circuit still, but IMO its actually cleaner to keep everything in one path - open to being persuaded otherwise though!

Reply:
but not short-circuit-ing results in having to add all those `if(!assetsOnly)` clauses, which add to the cognitive load IMO. I mean, it's not a whole lotta of extra load, but why not exit early if we can?

Reply:
basically this PR would become a one line change + tests no?

Reply:
Short circuit here feels simpler?
Otherwise there are more extra if statements below...

Reply:
yep sounds good i'll revert these then :)